abci/client/client.go

```diff
@@ -42,6 +43,7 @@ type Client interface {
 	SetOptionSync(types.RequestSetOption) (*types.ResponseSetOption, error)
 	DeliverTxSync(tx []byte) (*types.ResponseDeliverTx, error)
 	CheckTxSync(tx []byte) (*types.ResponseCheckTx, error)
+	RecheckTxSync(tx []byte) (*types.ResponseCheckTx, error)
```

Comment:
Good idea! Sync recheck tx makes sense if we already had a new tx in our mempool. To make this work we'd have to make this recheck share the same state as the normal check tx, which shouldn't be too complex to do. May want to save implementation of this for a separate pr

Reply:
Thanks for your review!! I found it's difficult to process the `checkTx`  function on the application. I will close this pull request and create a new pull request when I solve this issue completely.